### PR TITLE
Input validation (part 1)

### PR DESCRIPTION
**Pull request by kulmann**
This PR makes use of [ozzo-validation](https://github.com/go-ozzo/ozzo-validation) and sets up basic input validation for all requests. For now this focuses on having alphanumeric `identifier` values and account uuids being actual UUIDs and not creating folders on get or list requests. Plausibility checks on save requests for bundles and values will happen in a separate PR.

This fixes https://github.com/owncloud/ocis-settings/issues/15, https://github.com/owncloud/ocis-settings/issues/16 and https://github.com/owncloud/ocis-settings/issues/19

### CODE REVIEW

File: pkg/proto/v0/settings.pb.micro_test.go [hunk 1]
```diff
@@ -1123,53 +1239,3 @@ func TestSaveGetListSettingsValues(t *testing.T) {
 		})
 	}
 }
-
-//same as test above but List returns an empty array in case the extension name contains a `../`
```
Comment (individual-it) [inline pkg/proto/v0/settings.pb.micro_test.go:1127]: I cannot see where that test-case was added to the previous test, we should still test it

File: pkg/proto/v0/settings.pb.micro_test.go [hunk 2]
```diff
@@ -65,7 +80,12 @@ func TestSaveGetSettingsBundleWithNoSettings(t *testing.T) {
...
-			CustomError{},
+			CustomError{
+				ID:     "go.micro.client",
+				Code:   500,
+				Detail: "settings: cannot be blank.",
+				Status: "Internal Server Error",
+			},
 		},
 		{
 			"UTF",
```
[the remark this answers is not in the document]
Comment (refs) [reply pkg/proto/v0/settings.pb.micro_test.go:91]: Indeed, `displayName` should allow UTF8, and I think it does by looking at the validations: https://github.com/owncloud/ocis-settings/pull/22/files/86611f6e1c5bbd21f2e20ffbe7df2b1384748b02#diff-5a6d9a72bd4b635966f4bbc17df1e3e8R13

The only limitation is in alphanumeric characters on the bundle key. AFAIU displayName should allow any UTF8 character 🤔

[the remark this answers is not in the document]
Comment (refs) [reply pkg/proto/v0/settings.pb.micro_test.go:91]: The only condition for the displayName is that it is [present on save](https://github.com/owncloud/ocis-settings/pull/22/files/86611f6e1c5bbd21f2e20ffbe7df2b1384748b02#diff-5a6d9a72bd4b635966f4bbc17df1e3e8R34)

File: pkg/proto/v0/settings.pb.micro_test.go (hunk 1 above)
Comment (kulmann) [reply pkg/proto/v0/settings.pb.micro_test.go:1127]: The List request doesn't return an empty list anymore when the provided `extension` param is invalid. It returns a validation error, and that is already covered in the bundle properties test.

File: pkg/proto/v0/settings.pb.micro_test.go [hunk 3]
```diff
@@ -581,7 +641,18 @@ func TestGetSettingsBundleAccessOtherBundle(t *testing.T) {
 			BundleKey: "alice-bundle",
 		},
 		DisplayName: "alice settings bundle",
-		Settings:    nil,
+		Settings: []*proto.Setting{
```
[the remark this answers is not in the document]
Comment (kulmann) [reply pkg/proto/v0/settings.pb.micro_test.go]: yes, cool!

File: pkg/proto/v0/settings.pb.micro_test.go [hunk 4]
```diff
@@ -65,7 +80,12 @@ func TestSaveGetSettingsBundleWithNoSettings(t *testing.T) {
 			"simple-display-name",
 			"simple-extension-name",
 			"123e4567-e89b-12d3-a456-426652340000",
-			CustomError{},
```
[the remark this answers is not in the document]
Comment (kulmann) [reply pkg/proto/v0/settings.pb.micro_test.go]: done

File: pkg/proto/v0/settings.pb.micro_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kulmann) [reply pkg/proto/v0/settings.pb.micro_test.go]: The validation throws error for all invalid `identifier` properties (and sorts them by key). So it is fine to have all that should fail in one test.

[the remark this answers is not in the document]
Comment (kulmann) [reply pkg/proto/v0/settings.pb.micro_test.go]: I added a contract test that utf8 is allowed in bundle display names.

File: pkg/proto/v0/settings.pb.micro_test.go [hunk 5]
```diff
@@ -97,48 +125,65 @@ func TestSaveGetSettingsBundleWithNoSettings(t *testing.T) {
...
+			CustomError{
+				ID:     "go.micro.client",
+				Code:   500,
+				Detail: "bundle_key: must be in a valid format.",
+				Status: "Internal Server Error",
+			},
 		},
 		{
 			"spaces in values",
```
[the remark this answers is not in the document]
Comment (kulmann) [reply pkg/proto/v0/settings.pb.micro_test.go]: split the test into two (like with utf)

**Comment by kulmann** [comment]
@individual-it I addressed your comments, thanks a lot! Could you re-review?

**Pull request merged**